src/middlewares/createInsightsMiddleware.ts

```diff
@@ -91,27 +91,30 @@ export const createInsightsMiddleware: CreateInsightsMiddleware = (props) => {
     insightsClient('init', { appId, apiKey, ...insightsInitParams });
 
     const createWidget = connectConfigure(noop);
-    let widget: ReturnType<typeof createWidget>;
+    let configureClickAnalytics: ReturnType<typeof createWidget> | undefined;
+    let configureUserToken: ReturnType<typeof createWidget> | undefined;
 
     return {
       onStateChange() {},
       subscribe() {
         insightsClient('addAlgoliaAgent', 'insights-middleware');
 
-        let searchParameters: Partial<PlainSearchParameters> = {
-          clickAnalytics: true,
-        };
-        widget = createWidget({ searchParameters });
-        instantSearchInstance.addWidgets([widget]);
+        configureClickAnalytics = createWidget({
+          searchParameters: { clickAnalytics: true },
+        });
+        instantSearchInstance.addWidgets([configureClickAnalytics]);
 
         const setUserTokenToSearch = (userToken?: string) => {
-          instantSearchInstance.removeWidgets([widget]);
-          searchParameters = {
-            ...searchParameters,
-            userToken,
-          };
-          widget = createWidget({ searchParameters });
-          instantSearchInstance.addWidgets([widget]);
+          if (configureUserToken) {
+            instantSearchInstance.renderState[
+              instantSearchInstance.indexName
+            ].configure!.refine({ userToken });
```

Comment:
did you double check whether this left other parameters in place? I think it does, but not 100% sure

Reply:
yup, it's done by connectConfigure
https://github.com/algolia/instantsearch.js/blob/321b9521b9bbd3d6fb0dc861ac87e15ec966ba3c/src/connectors/configure/connectConfigure.ts#L101:L104